modules/scip/9.2.0.bcr.1/patches/scip.patch

```diff
@@ -0,0 +1,187 @@
+-   if ( ! WriteProtoToFile(filespec, proto, operations_research::ProtoWriteFormat::kProtoText, true) )
++   if ( ! WriteProtoToFile(filespec, proto, operations_research::ProtoWriteFormat::kProtoText, true).ok() )
+    {
+       SCIPerrorMessage("Could not write <%s>\n", fname);
+       return SCIP_READERROR;
+diff --git a/src/scip/config.h b/src/scip/config.h
+new file mode 100644
+index 0000000000..871fde8e55
+--- /dev/null
++++ b/src/scip/config.h
+@@ -0,0 +1,32 @@
++#ifndef __CONFIG_H__
++#define __CONFIG_H__
++
++#define CMAKE_BUILD_TYPE "Release"
++#define SCIP_VERSION_MAJOR 9
++#define SCIP_VERSION_MINOR 2
++#define SCIP_VERSION_PATCH 0
++#define SCIP_VERSION_SUB 0
++#define SCIP_VERSION_API 115
++/* #undef BMS_NOBLOCKMEM */
++/* #undef SCIP_NOBUFFERMEM */
++/* #undef WITH_DEBUG_SOLUTION */
++/* #undef SCIP_NO_SIGACTION */
++/* #undef SCIP_NO_STRTOK_R */
++/* #undef TPI_NONE */
++#define TPI_NONE
++/* #undef TPI_OMP */
++#define SCIP_THREADSAFE
++#define WITH_SCIPDEF
++/* #undef SCIP_WITH_LAPACK */
++/* #undef SCIP_WITH_PAPILO */
++#define SCIP_WITH_ZLIB
++/* #undef SCIP_WITH_READLINE */
++/* #undef SCIP_WITH_GMP */
++/* #undef SCIP_WITH_LPSCHECK */
++/* #undef SCIP_WITH_ZIMPL */
++/* #undef SCIP_WITH_AMPL */
++#define SCIP_ROUNDING_FE
++/* #undef SCIP_ROUNDING_FP */
++/* #undef SCIP_ROUNDING_MS */
++
++#endif
+diff --git a/src/scip/scip_export.h b/src/scip/scip_export.h
+new file mode 100644
+index 0000000000..871fde8e55
+--- /dev/null
++++ b/src/scip/scip_export.h
+@@ -0,0 +1,24 @@
++
++#ifndef SCIP_EXPORT_H
++#define SCIP_EXPORT_H
++
++#ifdef SCIP_STATIC_DEFINE
++#  define SCIP_EXPORT
++#  define SCIP_NO_EXPORT
++#else
++#  ifndef SCIP_EXPORT
++#    ifdef libscip_EXPORTS
++/* We are building this library */
++#      define SCIP_EXPORT
++#    else
++/* We are using this library */
++#      define SCIP_EXPORT
++#    endif
++#  endif
++
++#  ifndef SCIP_NO_EXPORT
++#    define SCIP_NO_EXPORT
++#  endif
++#endif
++
++#endif /* SCIP_EXPORT_H */
+diff --git a/src/scip/githash.c b/src/scip/githash.c
+new file mode 100644
+index 0000000000..d1e99c662d
+--- /dev/null
++++ b/src/scip/githash.c
+@@ -0,0 +1 @@
++#define SCIP_GITHASH "a740f0891e"
+\ No newline at end of file
+diff --git a/src/scip/scipbuildflags.c b/src/scip/scipbuildflags.c
+--- a/src/scip/scipbuildflags.c
++++ b/src/scip/scipbuildflags.c
+@@ -23,7 +23,6 @@
+ /* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+ /**@file   scipbuildflags.c
+- * @ingroup OTHER_CFILES
+  * @brief  build flags methods
+  * @author Felipe Serrano
+  */
+@@ -31,12 +30,36 @@
+ /*---+----1----+----2----+----3----+----4----+----5----+----6----+----7----+----8----+----9----+----0----+----1----+----2*/
+
+ #include "scip/scipbuildflags.h"
+-#include "scip/buildflags.h"
+
+ /** returns the flags that were used to build SCIP */
+ const char* SCIPgetBuildFlags(
+    void
+    )
+ {
+-   return SCIP_BUILDFLAGS "\n LPS=" SCIP_LPS "\n IPOPT=" SCIP_IPOPT;
++   return " ARCH=arm64\n\
```

Comment:
Could we patch this in buildflags.h instead?

Reply:
Done